# HG changeset patch
# User Rik <rik@octave.org>
# Date 1464904753 25200
#      Thu Jun 02 14:59:13 2016 -0700
# Node ID 25c8d554af3253c778b92eed6d19b02f6c619cb9
# Parent  b2f1f4cf13892e8ee7fc5995593d3dbbb4348002
# Parent  1b5285c20e0e2ed7fc7bd3cc8a1d52e3e80e6daa
maint: Periodic merge of stable to default.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1585,16 +1585,20 @@ function __gnuplot_draw_axes__ (h, plot_
              __do_enhanced_option__ (enhanced, hlgnd));
   else
     fputs (plot_stream, "unset key;\n");
   endif
   fputs (plot_stream, "set style data lines;\n");
 
   cmap = [cmap; addedcmap];
   cmap_sz += rows (addedcmap);
+  if (cmap_sz == 1)        # bug #48083, illegal one-element colormap
+    cmap = [cmap; cmap];
+    cmap_sz = 2;
+  endif
   if (length (cmap) > 0)
     fprintf (plot_stream,
              "set palette positive color model RGB maxcolors %i;\n",
              cmap_sz);
     fprintf (plot_stream,
              "set palette file \"-\" binary record=%d using 1:2:3:4;\n",
              cmap_sz);
     fwrite (plot_stream, [1:cmap_sz; cmap.'], "float32");
